src/components/marker.js

```diff
@@ -31,7 +31,12 @@ const propTypes = Object.assign({}, BaseControl.propTypes, {
   // Offset from the left
```

Comment:
I'm not in love with how much code ended up being required in here, but I still think it's a manageable size and broken down into small methods.

Reply:
> I'm not in love with how much code ended up being required in here, but I still think it's a manageable size and broken down into small methods.

How generic is this code? If another component wanted to also implement dragging would it be highly reusable. Could it make sense to organize it as

`Marker` > `DraggableControl` > `BaseControl`

It could look more appealing to have these methods in a separate class that is focused on dragging, rather than mixed with the Marker specific functionality.

Reply:
I like this idea, let me try restructuring it that way.